content/docs/concepts/low-power.md

```diff
@@ -0,0 +1,64 @@
+linkTitle: "Low Power"
+type: "docs"
+weight: 35
+description: >
+  How to consume less power on microcontrollers
+---
+
+Microcontrollers are usually rather non-demanding when it comes to power consumption.
+Couple of watts is the most you may expect a hobby project would use.
+
+Sometimes, that is not low enough.
+Power your project from a battery, and it'd benefit greatly from consuming less.
+
+Common scenario for such projects is sleeping most of the time, and only rarely briefly wake up to query sensors and execute actions.
+The main caveat here is if not configured properly, microcontroller and periferals consume power as usual while your program "sleeps".
+
+To address that, most microcontrollers have one or more "low power" modes, also known as "deep sleep" and, sometimes, "dormant".  
+In such modes microcontroller may consume as low as milli- or even micro-Watts.  
+
+## Implementations
+
+Manufacturers implement different modes, depending of capabilites of their products and name these modes differently too.
+Hence no naming consistency across microcontrollers, unfortunately.
+
+Always consult "Power consumption" part of a respective datasheet to understand capabilities of your hardware.  
+When reading datasheets, keep in mind peripherals that draw power too. These include \[power\] LEDs, various sensors or co-processors usually present on development boards.
+
+Capabilities of different chips:
+- Some very good chips (like the nRF series) can go to a very low power state (just a few µA) with RAM retention and waking up from a low-power RTC.
+- Some limited chips (like the esp32) can't really "sleep" with low power -- they can disable the core and memory and reset after a predefined time.
+- Other chips need to use a special feature to go into deep sleep mode.
+
+Especially the distinction between waking up with or without RAM is important, because it greatly affects how you write software.
+
+The common way to wake up from deep sleep is to use [interrupts](compiler-internals/interrupts), such as timer, real-time-clock or an interrupt on a pin change.
+
+### RP2040
+
+Minimal achievable power consumption according to [RP2040 datasheet](https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf): 6mW to 12mW (0.19mA at 3.3V)
+
+### nRF Family
+
+[nRF series chips](https://www.nordicsemi.com/Products) are exceptionally efficient and is recommended choice for low-power projects.
+
+Minimal achievable power consumption is in the area of micro-Watts.
+
+Even with BLE broadcasting enabled it can be lower power than rp2040 in its lowest power state!
+
+The things needed to make nrf series chips efficient are:
+- Disable the UART with help of `-serial=none` flag;
+- Use the special sleep function of the SoftDevice;
+
+## The Way to Low Power
+
+Following steps shall help you achieve low power consumption:
+
+- Put a for `{ arm.Asm("wfi") }` loop before anything else is initialized (at the beginning of `Reset_Handler`).
```

Comment:
Couple comments- `for` should be inside the code tildes. And maybe an explanation of what  `Reset_Handler` is? I've never used it.

Reply:
`Reset_Handler` is the "main's main" :D I don't know how to describe it best, probably just put the search link under the word?
https://github.com/tinygo-org/tinygo/search?q=%22export+Reset_Handler%22